### PR TITLE
Revamp modules structure

### LINKED ISSUE #233: Remove `Peekable`
### What's needed?

`Peekable` is a very niche and confusing interface, so it should be avoided.

### Proposed solution

Remove `Peekable`.

### Use cases

_No response_

### Alternatives and workarounds

_No response_

### Additional context

Depends on:

* https://github.com/frequenz-floss/frequenz-sdk-python/issues/772

### LINKED ISSUE #234: Remove `Bidirectional`
### What's needed?

Using bidirectional channels is not a recommended practice and should not be encouraged by this library.

### Proposed solution

Remove `Bidirectional`.

### Use cases

_No response_

### Alternatives and workarounds

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by llucax**
- Remove `Bidirectional` channel
- Remove `Peekable` and associated methods/classes
- Don't import `TypeVar`s from other modules
- Make all `TypeVar`s private
- Split `_base_classes` into `_receiver` and `_sender`
- Make channels return the base sender and receiver types
- Make channel's sender and receiver implementation private
- Don't use aliases for the base sender and receiver classes
- Move merge and select symbols to the top level
- Move utility receivers to their own public modules
- Move nested class out of the parent class

Fixes #233, fixes #234.

### CODE REVIEW

**Comment by llucax** [comment]
Added release notes, this is ready for the final review.

File: src/frequenz/channels/_broadcast.py [hunk 1]
```diff
@@ -176,18 +170,6 @@ def new_receiver(self, *, name: str | None = None, limit: int = 50) -> Receiver[
             recv.enqueue(self._latest)
         return recv
 
-    def new_peekable(self) -> Peekable[T]:
```
Comment (shsms) [inline src/frequenz/channels/_broadcast.py:179]: we should remove the `_latest` variable also from the `Broadcast` class.

**Comment by shsms** [comment]
Just one comment, looks good otherwise.

File: src/frequenz/channels/_broadcast.py (hunk 1 above)
Comment (llucax) [reply src/frequenz/channels/_broadcast.py:179]: No, we need it for the `resend_latest` option.

Comment (shsms) [reply src/frequenz/channels/_broadcast.py:179]: oh, right.

**Review by shsms — APPROVED** [review]

**Pull request merged**